ada/ast.py

```diff
@@ -8570,14 +8574,23 @@ def is_constant_object():
 
             # Renaming clause is:
             Entity.renaming_clause.then(
-                lambda rc: If(
+                lambda rc: Cond(
                     # Always constant if it renames a function_call
                     rc.renamed_object.is_call,
                     True,
+
+                    # Constant if the renamed object is an ExplicitDeref of an
+                    # access constant type.
+                    rc.renamed_object.is_a(T.ExplicitDeref),
+                    rc.renamed_object.is_constant,
+
                     # Constant if the renamed object is constant
                     rc.renamed_object.referenced_decl.is_constant_object
                 )
```

Comment:
Can't you just reuse the `Name.is_constant` property here on `rc.renamed_object`? It seems to me that this code is re-implementing some specific cases (but I may be wrong)

Reply:
This code was written before the `is_constant` property, explaining the duplicate implementation here. I fixed it, good catch!